### DISCUSSION TITLE
Cython: cimport pyproj.Geod

### DISCUSSION THREAD

**Discussion by daniprec** [pyproj4/pyproj#838] [Q&A]
Hello everyone!

I am currently developing a Python module that runs on Cython, and I want to use the class [pyproj.Geod](https://github.com/pyproj4/pyproj/blob/master/pyproj/geod.py).
To my knowledge, this class was build on Cython (via the hidden script _geod).
However, I cannot find a way to import the Geod class to my Cython class. Here is the code I am testing with:

```python
from pyproj import _geod
from pyproj cimport _geod


cdef class TestGeod:
    cdef _geod.Geod mygeod
    
    def __init__(self):
        pass

```

The error that arises when trying to compile is:

```
Error compiling Cython file:
------------------------------------------------------------
...
from pyproj import _geod
from pyproj cimport _geod
^
------------------------------------------------------------

./test_geod.pyx:2:0: 'pyproj.pxd' not found

Error compiling Cython file:
------------------------------------------------------------
...
from pyproj import _geod
from pyproj cimport _geod
^
------------------------------------------------------------

./test_geod.pyx:2:0: 'pyproj/_geod.pxd' not found

Error compiling Cython file:
------------------------------------------------------------
...
from pyproj import _geod
from pyproj cimport _geod


cdef class TestGeod:
    cdef _geod.Geod mygeod
        ^
------------------------------------------------------------
```

What is the proper way to import the Geod class to my custom Cython class?
Thanks in advance for your time!

**Comment by snowman2** [comment]
Currently, that isn't supported/tested by pyproj. But, you could submit a PR with that if you need it. Why are you not able to use the python class?

**Reply by daniprec** [reply]
To my knowledge, Cython classes don't support Python classes as properties. I want to build a Cython class that uses your pyproj.geod.Geod, which is a Python class.

I'll keep trying. If I find a right way to import it, I'll let you know.

**Reply by snowman2** [reply]
> To my knowledge, Cython classes don't support Python classes as properties

This should do the trick:
```
cdef readonly object mygeod
```

**Reply by daniprec** [reply]
It worked! Thanks for your help :)
